src/linear_algebra/dimension.lean

```diff
@@ -340,6 +340,14 @@ lemma exists_mem_ne_zero_of_dim_pos {s : submodule K V} (h : vector_space.dim K
   ∃ b : V, b ∈ s ∧ b ≠ 0 :=
 exists_mem_ne_zero_of_ne_bot $ assume eq, by rw [(>), eq, dim_bot] at h; exact lt_irrefl _ h
 
+lemma exists_mem_ne_zero_of_dim_pos' (h : vector_space.dim K V > 0) :
+  ∃ b : V, b ≠ 0 :=
```

Comment:
I guess the more direct way is to get a nonempty basis and use an element of that; I did this but it's not cleaner than your proof. I think going through the submodule lattice is fine. Also surprised this doesn't exist already, but it should be right here if it did...

Reply:
I [couldn't find](https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/Zero-dimensional.20vector.20space) the converse, either:
```
example (x : V) (h : vector_space.dim 𝕜 V = 0) : x = 0
```
How would one name that?

Reply:
Hmm. `vector_space.eq_zero_of_dim_eq_zero`? I'd swap the order of the arguments so `(x : V)` comes last. Then you can partially apply it to `h` to get a nice universal statement about `V`.

Reply:
... Turns out I found this converse rather tricky to prove -- I don't know enough about bases and cardinals.  Decided to leave it out rather than wrestling with it for longer.

Reply:
You're right, this is oddly tricky. Points to a few holes in the LA library, or at least corners of the library that I don't know about. I'll PR it once I clean it up.

Reply:
#3054, by @robertylewis, does both these facts; I will update this PR once the other is merged.